### ISSUE TITLE
@ts-ignore false positive inside strings and other comments

### ISSUE THREAD

**Issue by ajafff** [Microsoft/TypeScript#25240]
**TypeScript Version:**  3.0.0-dev.20180621

**Search Terms:** `ts-ignore`

**Code**

```ts
<>
    // @ts-ignore
    {nonExistent}
</>;

`
// @ts-ignore
${nonExistent}
`;

"\
// @ts-ignore"
nonExistent;

/**
// @ts-ignore all the things */
nonExistent;
```

**Expected behavior:**

3 times `Cannot find name 'nonExistent'.`

**Actual behavior:**

No error, because the regex used to lookup the comment doesn't (and cannot reliably) check the surroundings of the comment.

To fix this, you probably need to use `getTokenAtPosition` and `get(Leading|Trailing)CommentRanges` to check if this is really a comment.

**Playground Link:** https://agentcooper.github.io/typescript-play/?jsx=1#code/DwPgUABFEPQxABALgZwLQEsDmA7A9gE4Cmk0A3vjgKIAeGKSROSAvmMDCANxhgAGYOIlSZchEgBIKeanQZNW-HmABEAHUHxk6bPmIqwlWvUbNlMAFQXNwnWOIQAhgBtnEJAAsi7jxhxYUCAsYQxljeTMgA

**Related Issues:**

**Comment by mhegazy** [comment]
This is actually working as intended. `// @ts-ignore` is not meant to be a special lexical entity, it is just a marker on the line before, that can be insider another comment, JSX text or a string.
